sdk/file_sdk/Common/file_format_base.h

```diff
@@ -0,0 +1,36 @@
+#ifndef FILE_FORMAT_BASE_H
+#define FILE_FORMAT_BASE_H
+
+#include <string>
+#include <vector>
+#include "../../rmscrypto_sdk/CryptoAPI/IStream.h"
+#include "tag.h"
+using std::string;
+using std::vector;
+
+namespace mip{
+namespace file {
+
+class IFileFormat
+{
+public:
+  IFileFormat(rmscrypto::api::SharedStream& file, string extension);
+
+  virtual string GetOriginalExtension() = 0;
+
+  virtual void SetTags(vector<Tag>& tags);
+
+  virtual vector<Tag> GetTags();
+
+  virtual void Commit(rmscrypto::api::SharedStream& file, string& extension) = 0;

```

Comment:
I think the SharedStream is a bit misleading by hiding the word ptr. Let's use shared_ptr<IStream>.